src/mappings/staking.ts

```diff
@@ -278,42 +280,93 @@ async function updateStakeSummary(
       newDelegatorStake = amountBn;
     }
     const currEraIdx = await getCurrentEra();
-    const currEraId = currEraIdx.toString(16);
-    const prevEraIdx = currEraIdx - 1;
-    const prevEraId = prevEraIdx.toString(16);
+    const currEraId = BigNumber.from(currEraIdx).toHexString();
     const nextEraIdx = currEraIdx + 1;
-    const nextEraId = nextEraIdx.toString(16);
-
-    // FIXME - there is possibility that the indexer would quit and join stake at the same era (or at the next era), which would make some mistake in the current era and the next era
-    const isFirstStake =
-      !(await StakeSummary.get(prevEraId)) &&
-      !(await StakeSummary.get(currEraId));
+    const nextEraId = BigNumber.from(nextEraIdx).toHexString();
+
+    // update IndexerStakeSummary
+
+    let indexerStakeSummary = await IndexerStakeSummary.get(indexer);
+    let isFirstStake = false;
+
+    if (!indexerStakeSummary) {
+      isFirstStake = true;
+    } else if (
+      indexerStakeSummary.eraId === currEraId &&
+      indexerStakeSummary.totalStake === BigInt(0)
+    ) {
+      isFirstStake = true;
+    } else if (
+      indexerStakeSummary.eraId !== currEraId &&
+      indexerStakeSummary.nextTotalStake === BigInt(0)
+    ) {
+      isFirstStake = true;
+    }
 
-    if (isFirstStake) {
-      const currentStakeSummary = StakeSummary.create({
-        id: currEraId,
+    if (!indexerStakeSummary) {
+      indexerStakeSummary = IndexerStakeSummary.create({
+        id: indexer,
+        eraId: currEraId,
         totalStake: amountBn,
         indexerStake: newIndexerStake,
         delegatorStake: newDelegatorStake,
+        nextTotalStake: BigInt(0),
```

Comment:
next*Stake should be equal to current stake's value

Reply:
ok, I feel the logic is quite awk in my way. I will try to make it clearer.